### PR TITLE
Experiment: Stage 1 - Show All Steps At Once

### PR DESCRIPTION
**Pull request by andy1li**
image

**Checklist**:

- [x] I've thoroughly self-reviewed my changes
- [ ] I've added tests for my changes, unless they affect admin-only areas (or are otherwise not worth testing)
- [ ] I've verified any visual changes using Percy (add a commit with `[percy]` in the message to trigger)

## Summary by CodeRabbit

- **New Features**
	- Introduced conditional rendering for tutorial steps, allowing all steps to be displayed at once or in a navigable format based on user progress.
	- Added a help forum link that appears based on specific conditions to assist users.

- **Bug Fixes**
	- Improved logic for displaying completion status of steps to ensure accurate feedback.

- **Documentation**
	- Updated parameters in the tutorial card component to include current step information for better context.

### CODE REVIEW

File: app/components/course-page/course-stage-step/first-stage-tutorial-card.hbs [hunk 1]
```diff
@@ -1,42 +1,144 @@
...
+        </p>
+      </div>
+
+      <CopyableTerminalCommand @commands={{array 'git commit -am "[any message]"' "git push origin master"}} class="mb-3" />
+
+      <div class="prose dark:prose-invert">
+        <p class={{if this.hasPassedTests "line-through"}}>
+          Once you run the commands above, the
+          <code class="font-semibold text-red-700 bg-red-100 border border-red-200">Tests failed</code>
+          message below should be changed to
```
Comment (rohitpaulk) [inline app/components/course-page/course-stage-step/first-stage-tutorial-card.hbs:67]: @andy1li it looks like we're tweaking verbiage too here? This doesn't match the text we're using currently. 

Can you try to minimize the changes here? Small things can make big differences in experiments, ideally we test one aspect at a time

File: app/components/course-page/course-stage-step/first-stage-tutorial-card.hbs [hunk 2]
```diff
@@ -1,42 +1,144 @@
...
+            <p>Need help?</p>
+            <ul>
+              <li>
+                <a
+                  href="https://forum.codecrafters.io/new-topic?category=Challenges&tags=challenge%3A{{this.currentCourse.slug}}&title=%5B{{this.currentCourse.shortName}}%5D%20How%20to%20pass%20the%20first%20stage%3F&body=Checklist%3A%0A%0A1.%20%E2%9C%85%20or%20%E2%9D%8C%3A%20I%27ve%20uncommented%20the%20code.%0A2.%20%E2%9C%85%20or%20%E2%9D%8C%3A%20I%27ve%20saved%20the%20changes.%0A3.%20%E2%9C%85%20or%20%E2%9D%8C%3A%20I%27ve%20run%20the%20git%20commands%3A%0A%0A%60%60%60%0Agit%20commit%20-am%20%22%5Bseeking%20help%20on%20forum%5D%22%0Agit%20push%20origin%20master%0A%60%60%60%0A%0A---%0A%0AHere%E2%80%99s%20a%20screenshot%20showing%20the%20output%20from%20running%20the%20Git%20commands%3A%0A%0A%5BAttach%20screenshot%20here%5D%0A%0A%5BShare%20other%20details%20here%5D"
+                  target="_blank"
+                  rel="noopener noreferrer"
+                >
+                  Post your issue</a>
+                on the forum —
```
Comment (rohitpaulk) [inline app/components/course-page/course-stage-step/first-stage-tutorial-card.hbs:87]: @andy1li this was not supposed to be part of an experiment, no? It was supposed to apply to all users - again, don't see why this is coupled with this change

**Review by rohitpaulk — CHANGES_REQUESTED** [review]
Notes added

**Comment by andy1li** [comment]
image

File: app/controllers/course/stage/instructions.ts [hunk 3]
```diff
@@ -51,6 +55,10 @@ export default class CourseStageInstructionsController extends Controller {
     return !this.currentStep.courseStage.isFirst && this.currentStep.status === 'complete';
   }
 
+  get shouldShowNeedHelpForumLink() {
```
Comment (rohitpaulk) [inline app/controllers/course/stage/instructions.ts:58]:
```suggestion
  get shouldShowStage1ForumLinkCTA() {
```

**Review by rohitpaulk — APPROVED** [review]

**Pull request merged**